CMakeLists.txt

```diff
@@ -0,0 +1,17 @@
+cmake_minimum_required(VERSION 3.6)
+
+if (DEFINED ENV{VCPKG_ROOT})
+  set(CMAKE_TOOLCHAIN_FILE "$ENV{VCPKG_ROOT}/scripts/buildsystems/vcpkg.cmake")
+endif ()
+
+set(n clinfo)
+project(${n})
+
+find_package(OpenCL REQUIRED)
+
+add_executable(${n} src/${n}.c)
+target_include_directories(${n} PUBLIC src)
+target_include_directories(${n} PUBLIC ${OpenCL_INCLUDE_DIRS})
+target_link_libraries(${n} ${OpenCL_LIBRARIES})
```

Comment:
I think you need to link against CMAKE_DL_LIBS as well, if available. That's what the Makefile does.

Reply:
This is not needed.

Reply:
It is on Linux:
```bash
[2/2] Linking C executable clinfo
FAILED: : && /usr/lib/ccache/cc    CMakeFiles/clinfo.dir/src/clinfo.c.o  -o clinfo  /usr/lib/libOpenCL.so && :
/usr/bin/ld: error: undefined symbol: dlsym
>>> referenced by clinfo.c
>>>               CMakeFiles/clinfo.dir/src/clinfo.c.o:(oclIcdProps)
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
```

Reply:
You are of course correct :) I've added the change you suggest.